Lib/email/generator.py

```diff
@@ -186,8 +186,9 @@ def _write(self, msg):
         # If we munged the cte, copy the message again and re-fix the CTE.
         if munge_cte:
             msg = deepcopy(msg)
-            msg.replace_header('content-transfer-encoding', munge_cte[0])
             msg.replace_header('content-type', munge_cte[1])
+            if msg.get('content-transfer-encoding') is not None:
+                msg.replace_header('content-transfer-encoding', munge_cte[0])
```

Comment:
Can you set the header if it doesn't already exists?

See https://github.com/python/cpython/pull/1977#discussion_r163327866